### ISSUE TITLE
Partial method argument order

### ISSUE THREAD

**Issue by qqfunc** [beeware/rubicon-objc#453]
### What is the problem or limitation you are having?

As described in #26, when there is an Objective-C class that has `someMethod:arg1:arg2:` and `someMethod:arg2:arg1:` methods...

```python
>>> obj.someMethod(..., arg1=..., arg2=...)
>>> obj.someMethod(..., arg2=..., arg1=...)
```

These call the same method (`someMethod:arg1:arg2:`). This seems to be because partial methods only store a single order of arguments.

### Describe the solution you'd like

Allow them to be distinguished. (Then non-existent argument orders may no longer be allowed.) The argument names of partial methods are stored using dictionaries whose keys are `frozenset`. And, the values of the dictionaries are like `(selector string, [argument1 name, argument2 name, ...])`. In this case, storing multiple selectors and lists may be a solution.

### Describe alternatives you've considered

Store no argument names. Then, use [`respondsToSelector:`](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418583-respondstoselector) in `ObjCPartialMethod` to check if the method of specified name exists. (I feel there are some other ways...)

**Comment by freakboy3742** [comment]
Is this a problem you're having with an actual API? If so - which one? I'm a little surprised that Apple would construct an API where the argument order is the discriminator between method forms. 

Regardless, I think you've missed another possible option for implementation. ObjCPartial currently uses `frozenset` to track the options on an ObjCPartialMethod. If that were to be replaced with an "order preserving set" - i.e., more of a "tuple, but inserting the same item is prohibited" - then `arg1:arg2:` and `arg2:arg1` should hash differently, and thus yield different lookups. 

In fact, we might even be able to combine this with a fix for #148 - using a literal `tuple` would be a key part of that fix, with the other part being post-processing the kwargs to remove `__` suffixes.

**Comment by qqfunc** [comment]
I think this is only if I implement such methods myself. I understand such a situation won't be a problem in almost cases. Its priority would be low.

By the way, I thought the use of `frozenset` was for performance since the similar `list` is also contained in dictionary values. Is this wrong? Also, does the way of using `tuple` as keys mean that the `list` of the argument names is no longer necessary?

**Comment by freakboy3742** [comment]
The use of `frozenset` is partially for performance; but at least as currently implemented, it's also being used for the uniqueness properties of `set`. At the time that code was introduced, dictionary order (and thus kwargs order) couldn't be guaranteed. As of Python 3.6, we can now guarantee kwargs order, so that isn't a concern.

And yes, the `list` is serving almost exactly the same purpose, so there's very likely an optimisation to be had simplifying `ObjCPartialMethod.methods` to be a simple lookup of `order->name`, rather than `frozenset(order)->(name, order)`.

Interestingly, [the code includes a reference to the exact behavior you're describing here](https://github.com/beeware/rubicon-objc/blob/main/src/rubicon/objc/api.py#L1043) - along with a TODO to clean it up once Python 3.6 is deprecated :-)

**Comment by qqfunc** [comment]
Thank you. That seems to be true.

**Comment by freakboy3742** [comment]
Fixed by #462.
